components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.hpp

```diff
@@ -107,8 +107,12 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
         qw(i,k) = qv(i,k) + qc(i,k);
 
         // Temperature
+        // NOTE: vtheta (thv) is inconsistent with one in HOMME
+        // SEE gh issue #2813 and gh pr #2812
         const auto theta_zt = PF::calculate_theta_from_T(T_mid(i,k),p_mid(i,k));
+        // TODO? Can use PF::calculate_thetal_from_theta instead? (added in gh pr #2812)
         thlm(i,k) = theta_zt-(theta_zt/T_mid(i,k))*(latvap/cpair)*qc(i,k);
+        // TODO? Can use PF::calculate_virtual_temperature instead? (also be consistent with HOMME)
```

Comment:
I agree, 1st one would be BFB and should be used if the function now exist. @bogensch will have to answer the 2nd one.

Just for context, this was ported from https://github.com/E3SM-Project/scream/blob/a73d48a5f8556e5240b64b037bc60d42cb5f2413/components/eam/src/physics/cam/shoc_intr.F90#L774 which is a preprocess to shoc_main().

Reply:
@mahf708 okay, I see where the confusion is (and sorry for not catching it before... Monday mornings are hard)... It's that the SHOC definition of virtual temperature has liquid water loading vs the function call which does not.  I just saw the Slack discussion debating about whether or not this was a bug.  It is not a bug, it was intentional.  Moist boundary layer parameterizations often do including the liquid water loading in the definition (see bottom of page 15-7 of https://atmos.uw.edu/academics/classes/2013Q2/547/lect15.pdf and https://slideplayer.com/slide/13265361/).  

Thus, I retract my earlier statement about supporting using the Tv function here if it does not include liquid water loading since I'm not sure why the effect of doing this would be at this time.